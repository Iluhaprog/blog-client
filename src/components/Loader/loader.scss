.loader {
    opacity: 0;
    transition: .01s ease-in-out;
    margin: 5px 15px;

    &_visible {
        opacity: 1;
    }
}