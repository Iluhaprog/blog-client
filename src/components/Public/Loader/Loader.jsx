import React from 'react';
import styled from 'styled-components';

const Box = styled.div`
  position: fixed;
  width: 100vw;
  height: 100vh;
  display: flex;
  justify-content: center;
  align-items: center;

  .spinner-block {
    width: 125px;
    text-align: center;
    margin-right: 0;

    .spinner {
      position: relative;
      width: 125px;
      height: 125px;

      &:before,
      &:after {
        content: "";
        display: block;
        position: absolute;
        border-width: 4px;
        border-style: solid;
        border-radius: 50%;
      }
    }
    
    .spinner.spinner-3 {

      @keyframes scale-2 {
        0% {
          transform: scale(0);
          opacity: 0;
        }

        50% {
          transform: scale(0.7);
          opacity: 1;
        }

        100% {
          transform: scale(1);
          opacity: 0;
        }
      }

      &:before {
        width: 125px;
        height: 125px;
        border-color: #000;
        top: 0;
        left: 0;
        animation: scale-2 1s linear 0s infinite;
      }

      &:after {
        width: 125px;
        height: 125px;
        border-color: #000;
        top: 0;
        left: 0;
        opacity: 0;
        animation: scale-2 1s linear 0.5s infinite;
      }
    }

  }
`;

const Loader = () => {
  return (
    <Box>
      <div className={'spinner-block'}>
        <div className={'spinner spinner-3'}>
        </div>
      </div>
    </Box>
  );
};

export {Loader};
