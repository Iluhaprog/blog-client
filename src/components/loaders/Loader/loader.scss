.loader {
    opacity: 0;
    transition: .3s ease-in-out;

    &_visible {
        opacity: 1;
    }
}