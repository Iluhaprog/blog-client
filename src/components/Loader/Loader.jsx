import React from 'react';
import loader from '../../assets/loader.svg';
import './loader.scss';

export default props => (
    <>
        <img 
            src={loader}
            width={props.width || 30}
            height={props.height || 30} 
            className={`loader ${props.visible ? 'loader_visible' : ''}`}
            alt='loading...'     
        />
    </>
);