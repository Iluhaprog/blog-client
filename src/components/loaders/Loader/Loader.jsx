import React from 'react';
import PropTypes from 'prop-types';
import './loader.scss';

function Loader(props) {
    const isVisible = props.visible ? 'loader_visible' : '';
    return (
        <div className={`loader ${isVisible} ${props.className}`}>
            {props.children}
        </div>
    );
}

Loader.defaultProps = {
    visible: false,
};

Loader.propTypes = {
    visible: PropTypes.bool,
    children: PropTypes.node,
};

export default Loader;