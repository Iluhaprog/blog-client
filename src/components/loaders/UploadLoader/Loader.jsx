import React from 'react';
import PropTypes from 'prop-types';
import loader from '../../../assets/loader.svg';
import './loader.scss';

function Loader(props){
    return (
        <>
            <img 
                src={loader}
                width={props.width || 30}
                height={props.height || 30} 
                className={`loader ${props.visible ? 'loader_visible' : ''}`}
                alt='loading...'     
            />
        </>
    );
}

Loader.defaultProps = {
    width: 30,
    height: 30,
    visible: false,
};

Loader.propTypes = {
    width: PropTypes.number,
    height: PropTypes.number,
    visible: PropTypes.bool,
};

export default Loader;