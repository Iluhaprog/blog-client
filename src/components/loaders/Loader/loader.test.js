import React from 'react';
import { render } from '@testing-library/react';
import Loader from './Loader';

test('Should render loader component (visible false)', () => {
    const { asFragment } = render(<Loader>Loading...</Loader>);
    expect(asFragment()).toMatchSnapshot();
});

test('Should render loader component (visible true)', () => {
    const { asFragment } = render(<Loader visible={true}>Loading...</Loader>);
    expect(asFragment()).toMatchSnapshot();
});