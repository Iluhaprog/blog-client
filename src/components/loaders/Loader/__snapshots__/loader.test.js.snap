// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render loader component (visible false) 1`] = `
<DocumentFragment>
  <div
    class="loader "
  >
    Loading...
  </div>
</DocumentFragment>
`;

exports[`Should render loader component (visible true) 1`] = `
<DocumentFragment>
  <div
    class="loader loader_visible"
  >
    Loading...
  </div>
</DocumentFragment>
`;
